### ISSUE TITLE
GraphQL returns 500 errors.

### ISSUE THREAD

**Issue by Pinski** [Yelp/yelp-fusion#270]
Hello,

So in the past day or two, the GraphQL end point has started to return 500 errors.  I've tried various queries directly against the API in Postman and cannot seem to get a response besides 500.  All of my code that called Yelp was working fine a few days ago and nothing has changed since then.  In fact, I code still correctly calls other endpoints in Yelp like search.

I've double checked that my authentication token is correct, that I'm using the correct GraphQL queries from Yelp examples, and that my app is still in the developer beta, everything seems fine on my end.  

Here are some of the queries that have failed for me:
```
{
    b1: business(id: "yelp-san-francisco") {
        name
        id
        rating
        review_count
        photos
    }
    b2: business(id: "garaje-san-francisco") {
        name
        id
        rating
        review_count
        photos
    }
}
```

```
{
    b1: business(id: "yelp-san-francisco") {
        ...basicBizInfo
    }
    b2: business(id: "garaje-san-francisco") {
        ...basicBizInfo
    }
}

fragment basicBizInfo on Business {
  id
  name
  url
  rating
  review_count
  categories {
      title
      alias
  }
  location {
      address1
      address2
      address3
      city
      state
      zip_code
  }
  display_phone
  coordinates {
      latitude
      longitude
  }
  hours {
      is_open_now
  }
}
```

```
{ 
b1: business(id: "de-afghanan-cuisine-fremont") { 
    ...businessInfo 
} 
b2: business(id: "de-afghanan-kabob-house-fremont-3") { 
    ...businessInfo 
} 
b3: business(id: "de-afghanan-kabob-house-san-francisco-2") { 
    ...businessInfo 
} 
b4: business(id: "gulzaar-halal-restaurant-and-catering-san-jose-7") { 
    ...businessInfo 
} 
b5: business(id: "helmand-palace-san-francisco") { 
    ...businessInfo 
} 
b6: business(id: "kabob-trolley-san-francisco-5") { 
    ...businessInfo 
} 
b7: business(id: "kabul-afghan-cuisine-san-carlos") { 
    ...businessInfo 
} 
b8: business(id: "kabul-afghan-cuisine-sunnyvale-2") { 
    ...businessInfo 
} 
b9: business(id: "kabul-express-kabob-newark") { 
    ...businessInfo 
} 
b10: business(id: "kabul-kabob-and-grill-dublin") { 
    ...businessInfo 
} 
} 

fragment businessInfo on Business { 
    
id
photos
name
url
rating
review_count
price
categories {
    title
    alias
}
location {
    address1
    address2
    address3
    city
    state
    zip_code
}
display_phone
coordinates {
    latitude
    longitude
}
hours {
    is_open_now
}

} 

```

Here's the response they all get:
```
<html>
    <head>
        <title>500 Internal Server Error</title>
    </head>
    <body>
        <h1>500 Internal Server Error</h1>
  The server has either erred or is incapable of performing the requested operation.
        <br/>
        <br/>
    </body>
</html>
```

Any thoughts?

**Comment by tomelm** [comment]
Hey @Pinski! Sorry about the breakage, I recently pushed out a change to how the endpoint handles requests. Could you add the following content type header and see if it works better for you: `Content-Type: application/graphql`.

**Comment by Pinski** [comment]
Hey, no worries about the breakage.  You've got to do what you've got to do.  Setting the Content-Type to differentiating between JSON and GraphQL makes a lot of sense.

Unfortunately, adding `Content-Type: application/graphql` didn't fix it.  Now I'm getting 400 errors:

```
{
    "error": {
        "code": "CLIENT_ERROR",
        "description": "Bad Request"
    }
}
```

I'm sending the basic example on the page you linked:

```
{
    business(id: "garaje-san-francisco") {
        name
        id
        rating
        url
    }
}
```

I know my token is working correctly because I removed it and put in a garbage value and got different errors.  I'm sure I'm sending `Content-Type: application/graphql` too.

Anything else I can try?

**Comment by tomelm** [comment]
Hmm, so that query works for me okay. How are you sending the request in? curl or code or something else?

**Comment by Pinski** [comment]
I only tested it in Postman last night.  Maybe I'm missing something simple.

![image](https://user-images.githubusercontent.com/1851911/28685580-1a86567c-72bd-11e7-9b15-a1f9bafc51e9.png)
![image](https://user-images.githubusercontent.com/1851911/28685608-3323fb58-72bd-11e7-99c8-ea4df333014b.png)

I just tried it in my .NET code and setting the correct Content-Type fixed it in code.  I was mostly worried about my code not working, so at least that's fix.  Still not sure why it's not working in Postman though.

**Comment by tomelm** [comment]
Hmm, unfortunately I'm having trouble reproducing this. I think one of my coworkers was seeing an issue with this earlier today, I'll try to reproduce this with them tomorrow.

**Comment by tomelm** [comment]
Hey @Pinski  are you or postman setting any Accept-Language headers? I debugged it here and the error was because a bad locale was being provided.

**Comment by Agheb** [comment]
I had the same issues with Postman for Chrome. The native version for MacOS worked flawlessly

**Comment by Pinski** [comment]
@tomelm so I just tested it again and it wasn't working correctly.  Then I updated my installation of Postman and now it does work.  I don't think Postman was sending any Accept-Language headers, but I can't be sure.  I know it is not sending them now.

I noticed, after updating, that Postman now auto-fills in application/graphql when I started typing it in the Content-Type.  Maybe Postman didn't "officially" support GraphQL or send the correct headers before.  I don't know.

So it looks like if you update Postman to latest, it now works.  Sorry you spent time debugging this.  Just FYI, I'm using the Windows Native App v5.1.2 now.
